# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ModuleLoader.h
# Commit: 0ac6d9d26e2b
# Full Hash: 0ac6d9d26e2ba97e31eb1f1a0e75d93071c2cf50
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-18 09:08:22
# Regressor Bug: 1742437
# File Overlap Count: 1
# Description:
#   Bug 1742437 - Split ScriptLoadRequest into ScriptLoadRequest and DOMScriptLoadContext; r=jonco,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D134041
# ==============================================================================

diff -r 5a00e56ade15 -r 0ac6d9d26e2b dom/script/ModuleLoader.h
--- a/dom/script/ModuleLoader.h	Thu Feb 17 15:45:19 2022 +0000
+++ b/dom/script/ModuleLoader.h	Thu Feb 17 15:57:48 2022 +0000
@@ -70,6 +70,10 @@
   virtual void EnsureModuleHooksInitialized() = 0;
   virtual nsresult StartModuleLoad(ScriptLoadRequest* aRequest) = 0;
   virtual void ProcessLoadedModuleTree(ModuleLoadRequest* aRequest) = 0;
+  virtual nsresult CompileOrFinishModuleScript(
+      JSContext* aCx, JS::Handle<JSObject*> aGlobal,
+      JS::CompileOptions& aOptions, ModuleLoadRequest* aRequest,
+      JS::MutableHandle<JSObject*> aModuleScript) = 0;
 };
 
 class ModuleLoader : public nsISupports {